### PR TITLE
Improve error message for FE model export checking

### PR DESCRIPTION
**Pull request by BowenBao**

### CODE REVIEW

**Review by thiagocrepaldi — APPROVED** [review]

**Review by thiagocrepaldi — CHANGES_REQUESTED** [review]
Could you also improve the help message at orttrainer_options.py? 

```
debug.check_model_export (bool, default is False)
            compares PyTorch outputs with InferenceSession outputs before the first
            train step to ensure successful model export
```

It mentions InferenceSession, which is an implementation detail. We should inform the check is done by doing inference in the model using pytorch and ONNX, but not list internal classes, etc on the help message

File: orttraining/orttraining/python/training/orttrainer.py [hunk 1]
```diff
@@ -285,19 +285,19 @@ def save_as_onnx(self, path):
         with open(path, "wb") as f:
             f.write(self._onnx_model.SerializeToString())
 
-    def _debug_model_export(self, input):
+    def _check_model_export(self, input):
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/python/training/orttrainer.py:288]: Should it be "check" or "verify"?

**Comment by thiagocrepaldi** [comment]
I noticed that the code only look for 'Dropout' nodes, but not 'TrainableDropout'. Is that expected?

**Comment by thiagocrepaldi** [comment]
Not directly related to this change, but to the feature itself, on PyTorch documentation, they mention:

```
Remember that you must call model.eval() to set dropout and batch normalization layers to evaluation mode before running inference. Failing to do this will yield inconsistent inference results.
```

Should we also capture batch normalization layers and put them in evaluation mode to remove randomness?

File: orttraining/orttraining/python/training/orttrainer.py (hunk 1 above)
Comment (BowenBao) [reply orttraining/orttraining/python/training/orttrainer.py:288]: i don't have a strong preference over the two. the update is to have it consistent with the flag.

**Comment by BowenBao** [comment]
@thiagocrepaldi 

`TrainableDropout` is deprecated. It is not an official ONNX op, so won't be produced by the exporter.

Synced offline, i think it is fine as it does not affect our case in checking. When mode left on training, bn is going to produce different outputs for subsequent runs, but the first run should stay the same.

**Review by thiagocrepaldi — APPROVED** [review]
